tools/wt_cmp_uri

```diff
@@ -0,0 +1,308 @@
+    print('  uri1 and uri2 are WiredTiger URIs, e.g. table:foo or file:abc.wt')
+    print('Options:')
+    print('  -v      verbose - shows every key/value')
+    print('  -t ts   read at a timestamp (a different timestamp for each dir may be specified)')
+    sys.exit(1)
+
+verboseFlag = False
+def verbose(s):
+    if verboseFlag:
+        print("VERBOSE>>> " + s)
+
+# An encapsulation of a WT cursor
+class CompareCursor:
+    def __init__(self, cursor, uri, name, reverse):
+        self.cursor = cursor
+        self.uri = uri
+        self.name = name
+        self.reverse = reverse
+
+    def close(self):
+        session = self.cursor.session
+        self.cursor.close()
+        session.close()
+
+# Open a cursor at a timestamp
+def wt_open(conn, timestamp, uri):
+    session = conn.open_session()
+    if timestamp != None:
+        config = 'read_timestamp=' + timestamp
+        verbose('begin_transaction config: ' + config)
+        session.begin_transaction(config)
+    return session.open_cursor(uri, None, 'readonly')
+
+# Find a uri in the metadata and determine if it uses the 'reverse' collator.
+# test/format may use this collator.
+def is_reverse(session, uri):
+    mc = session.open_cursor('metadata:')
+    config = mc[uri]
+    mc.close()
+    return 'collator=reverse' in config
+
+# Get an encapsulation of a WT cursor at the given timestamp.
+def get_compare_cursor(conn, timestamp, uri, arg):
+    cursor = wt_open(conn, timestamp, uri)
+    reverse = is_reverse(cursor.session, uri)
+    return CompareCursor(cursor, uri, arg, reverse)
+
+# Show a limited length version of a key/value
+def show_kv(kv):
+    s = str(kv)
+    if len(s) > 100:
+        s = s[0:100] + '...'
+    return s
+
+# Walk through both cursors, comparing them.  Generally, it's very easy, when the
+# keys don't compare equally, we would know to do 'next' on the smaller key.  Except
+# when custom collation is behind, we don't really know how to 'recover' in the general case.
+# test/format uses custom collation, but only with a "reverse" collator - if we detect keys
+# are going in reverse, we take note of that.
+# 
+def compare_cursors(cc1, cc2, version):
+    nrecords = 0
+    verbose('{}: next'.format(cc1.name))
+    if cc1.reverse != cc2.reverse:
+        print('ERROR: different collators for URIs {} and {}'.format(cc1.name, cc2.name))
+        return
+    reverse = cc1.reverse
+    while cc1.cursor.next() == 0:
+        k1 = cc1.cursor.get_key()
+        v1 = cc1.cursor.get_value()
+        verbose('  key: {} value: {}'.format(show_kv(k1), show_kv(v1)))
+
+        verbose('{}: next'.format(cc2.name))
+        if cc2.cursor.next() != 0:
+            print('EOF on {}'.format(cc2.name))
+            break
+        nrecords += 1
+        k2 = cc2.cursor.get_key()
+        v2 = cc2.cursor.get_value()
+        verbose('  key: {} value: {}'.format(show_kv(k2), show_kv(v2)))
+
+        missing = 0
```

Comment:
Why is `missing` getting sometimes negative and sometimes positive? Would it make sense to have one missing count, which is a sum of missing keys across both tables, or two different missing counts for the two different tables?